index.js

```diff
@@ -100,62 +100,70 @@ function getStylesFromIframe(iframe) {
 }
 
 module.exports = function() {
+    var codepens = document.querySelectorAll(".codepen");
+    var parents = [];
+    //remove the old codepen links
+    codepens.forEach(function(codepen, i){
+        var el = findSelector(codepen, 'pre, .demo_wrapper');
+        parents.push(el);
+        codepen.parentNode.removeChild(codepen);
+    });
 
-    document.body.addEventListener("click", function(ev){
-        if(matches.call(ev.target, ".codepen")){
-
-            var el = findSelector(ev.target, "pre, .demo_wrapper");
-            if(el && matches.call(el, "pre")) {
-                var preElement = el;
-                var codeElement = preElement.querySelector("code");
-                var language = codeElement.className.match(languageHTML)[1];
-                var text = codeElement.textContent;
+    //Register PrismJS "Run" custom button
+    Prism.plugins.toolbar.registerButton("run-code", function(env) {
+        var btn = document.createElement("button");
+        btn.innerHTML = "Run";
+        btn.addEventListener('click', function() {
+            var el;
+            for (var i = 0; i < parents.length; i++) {
+                if (parents[i].contains(env.element)) {
+                    el = parents[i];
+                    break;
+                }
+            };
```

Comment:
Does `registerButton` only need to be called once (vs. a call for each instance of the toolbar)?

Instead of keeping the `parents`, should we walk up from `env.element` to find the `pre`?

Reply:
The `registerButton` needs to be called once otherwise it register all the buttons for each of the codes.

For `parents` I did try to walk up but it didn't work.